Orange/widgets/widget.py

```diff
@@ -738,6 +745,32 @@ def _userconfirmed():
 
         self.__msgwidget.accepted.connect(_userconfirmed)
 
+    @classmethod
+    def migrate_settings(cls, settings, version):
```

Comment:
If the settings break with a change, so might the widget that uses the component. If for instance a component used to store a list of indices in a setting and now it stores a single number, a widget trying to index the appropriate component attribute will crash.

While I can move the migrate_settings method to the component, what can we do with migrate_context? Leave it on the widget? If this is the case, I would prefer to keep the methods in the same place. And if the migrations are placed on a component, who defines the settings version?

IMO, if component is complicated enough to warrant its own migration, it can define its own migrate_settings, migrate_context, but the widget should be aware of that and call the method explicitly in its own migrate_settings method with proper parameters.